apps/nextjs-website/src/AppEnv.ts

```diff
@@ -0,0 +1,47 @@
+import { pipe } from 'fp-ts/lib/function';
+import * as E from 'fp-ts/lib/Either';
+import { Config, makeConfig, publicEnv } from './config';
+import { WebinarEnv } from './lib/webinars/webinarQuestions';
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
+import { makeAwsCredentialsFromCognito } from './lib/makeAwsCredentialsFromCognito';
+import { Auth } from 'aws-amplify';
+
+// This type represents the environment of the application. It contains
+// configuration as well as other dependencies required by the application. In
+// other words contains all runtime configuration and global functions that may
+// be mockable
+export type AppEnv = {
+  readonly config: Config;
+} & WebinarEnv;
+
+// given environment variables produce an AppEnv
+const makeAppEnv = (
+  env: Record<string, undefined | string>
+): E.Either<string, AppEnv> =>
+  pipe(
+    makeConfig(env),
+    E.map((config) => ({
+      config,
+      questionLifetimeInSeconds:
+        config.NEXT_PUBLIC_WEBINAR_QUESTION_LIFETIME_IN_SECONDS,
+      nowDate: () => new Date(),
+      dynamodbClient: new DynamoDBClient({
```

Comment:
```suggestion
      dynamoDbClient: new DynamoDBClient({
```
wdyt?

Reply:
I would go with `dynamodbClient` or `dynamoDBClient` 😄, which one do you prefer?

Reply:
`dynamoDBClient` 😄